drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -457,11 +500,12 @@ def _is_finished(self):
         completed_count = 0
         total_count = 0
         item_state = self._get_item_state()
+
         for item in self.data['items']:
-            if item['zone'] != 'none':
+            item_id = item['id']
+            if self._get_valid_item_zones(item_id) != []:
                 total_count += 1
-                item_id = str(item['id'])
-                if item_id in item_state:
+                if str(item_id) in item_state:
                     completed_count += 1
```

Comment:
Found a way to go with this that preserves current behavior and gives us a comprehensive starting place for the future.